### ISSUE TITLE
Companion install problem + FF 59.0.2 64bits crashes on Ubuntu 17.10

### ISSUE THREAD

**Issue by liar666** [mi-g/vdhcoapp#24]
Hi,

My context:
```
$ lsb_release -a
...
Description:    Ubuntu 17.10
Codename:       artful
...

# apt-get dist-upgrade
...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

What I did:

- Trying to install .deb
```
# dpkg -x net.downloadhelper.coapp-1.1.3-1_amd64.deb 
Unpacking net.downloadhelper.coapp (1.1.3) ...
dpkg: error processing archive net.downloadhelper.coapp-1.1.3-1_amd64.deb (--install):
 unable to open '/etc/opt/chrome/native-messaging-hosts/net.downloadhelper.coapp.json.dpkg-new': No such file or directory
Errors were encountered while processing:
 net.downloadhelper.coapp-1.1.3-1_amd64.deb
```
RESULT: not installed
- Trying to install .tar.gz with included scripts
```
# tar xvzf net.downloadhelper.coapp-1.1.3-1_amd64.tar.gz
# cd net.downloadhelper.coapp-1.1.3
# ./setup-linux-system.sh 
Executable file net.downloadhelper.coapp-linux-64 is not in the current directory
# ./setup-linux-user.sh 
Executable file net.downloadhelper.coapp-linux-64 is not in the current directory
```
RESULT: not installed
- Installed .tar.gz as system with commands from (https://www.downloadhelper.net/install-coapp)
```
# tar xvzf net.downloadhelper.coapp-1.1.3-1_amd64.tar.gz
# cd net.downloadhelper.coapp-1.1.3/bin
# ./net.downloadhelper.coapp-linux-64 install --system
VdhCoApp: VdhCoApp is ready to be used
```
RESULT: install worked, Addons pref pages displays "Found companion app: VdhCoApp 1.1.3[...]"
Trying to download 2 videos(*) resulted in FF crash.
- Installed .tag.gz as user with commands from https://www.downloadhelper.net/install-coapp
```
$ ./net.downloadhelper.coapp-linux-64 install --user
VdhCoApp: VdhCoApp is ready to be used
```
RESULT: install worked, Addons pref pages displays "Found companion app: VdhCoApp 1.1.3[...]"
Trying to download same 2 videos(*) resulted in FF crash.

Is there a way to find some logs that would explain the crashes?

(*) I had a few tabs opened in 2 windows plus a 3rd tab with the 2 following videos to be downloaded (in the lowest resolution)
https://labs.spotify.com/2014/03/27/spotify-engineering-culture-part-1/
https://labs.spotify.com/2014/09/20/spotify-engineering-culture-part-2/

**Comment by liar666** [comment]
UPDATE:
I tried again to download the videos **one after the other**:
- First video, I did the download while the video was playing => OK
- Second video, I did the download with the video paused => FF crashed
I'm still not sure if the problem is from video2 or with paused/unpaused videos

**Comment by mi-g** [comment]
We recommend using `dpkg -i net.downloadhelper.coapp-1.1.3-1_amd64.deb`, not `dpkg -x ...`.

The VDH coapp is unlikely to be the cause of the Firefox crash (otherwise it's a big bad bug in Firefox, but i haven't heard about such reports). Most of the times, Firefox crashes are due to the browser trying to write somewhere on the disk and the system fails to do so in a bad way (quite often a disk/mounted-filesystem issue) causing the main app to crash.

**Comment by liar666** [comment]
Hi, Thanks for the quick reply.

FYI, I tried with second video only, while playing it. It worked until 95% when it crashed at ~35% when paused, but then it crashed also :{

Sorry, typing error, I indeed used `dpkg -i` not `dpkg -x`...

VDH is supposed to write on ~/dwhelper, which is local and where I have 148GB free...

**Comment by mi-g** [comment]
The vdh coapp was developed and tested on Ubuntu 16.04 (latest ESR), i cannot tell why you get this error on 17.10, i haven't heard of it before. Very soon, a new ESR will be available (18.04) and the procedure will be rechecked and fixed where required. Hope fully, this will solve the 17.10 issue.

It's important to know if the downloads are made by the coapp or the browser: if it's a "complex" (YouTube, DASH, HLS, ...), this is done by the coapp (due to new limited Firefox Quantum capabilities), if it's "simple" (1 URL to 1 File) you are asked for browser or coapp (you can change this from the VDH settings). If it is using the coapp, even if there is a crash of the app, it should be on the coapp not Firefox.

Note you may have sector errors on the local disk, and if you are downloading using the browser, this may cause the crash.

**Comment by mi-g** [comment]
Are you having the same problem with [vdhcoapp 1.2.1](https://github.com/mi-g/vdhcoapp/releases/tag/v1.2.1) ?
